en/advanced_config/ethernet_setup.md

```diff
@@ -0,0 +1,208 @@
+   ```
+
+1. Copy and paste the following configuration information into the file (note: the indentations are important!):
+   ```
+   network:
+     version: 2
+     renderer: NetworkManager
+     ethernets:
+         enp2s0:
+             addresses:
+                 - 192.168.0.1/24
+             nameservers:
+                 addresses: [192.168.0.1]
+             routes:
+                 - to: 192.168.0.1
+                   via: 192.168.0.1
+   ```
+   
+   Save and exit the editor.
+1. Apply the *netplan* configuration by entering the following command into the Ubuntu terminal.
+   ```
+   sudo netplan apply
+   ```
+   
+If you have already configured the Pixhawk 5X, at this point you could connect it to the computer via an Ethernet cable and it would be automatically deteced by QGroundControl.
+  
+### Companion Computer Ethernet Network Setup
+
+The setup for a companion computer will depend on the companion computer operating system.
+A Linux operating system may support netplan, in which case the instructions would be the same as above, but using a unique IP address.
+
+<!-- Then link or continue on with MAVSDK setup - depending on what works when you try it -->
+
+## MAVSDK Sanity check
+
+Since we have to listen to port 14550 as mentioned earlier, the only thing changes in MAVSDK is the port we connect to UDP. 
+
+### Example for MAVSDK-Python:
+
+To give an idea about how this can be done, we make a change in a MAVSDK-Python example in its Github repo. You can extend this to your own application accordingly. Change [this line](https://github.com/mavlink/MAVSDK-Python/blob/3c2ad41273a42597499d5c4ec4ad90c6a494e691/examples/telemetry.py#L10) to the one below in your local script:
+```
+    await drone.connect(system_address="udp://:14550")
+```
+Successfully connected one gives you Telemetry information from your flight controller. 
+
+### Example for ROS2:
+
+Prerequisites:
+
+- You have a supported autopilot hardware with RTPS feature enabled firmware on it by using [this guide](https://docs.px4.io/master/en/middleware/micrortps.html#client-px4-px4-autopilot).
+- [ROS2](https://docs.px4.io/master/en/ros/ros2_comm.html#sanity-check-the-installation) has been set up correctly and [sanity check](https://docs.px4.io/master/en/ros/ros2_comm.html#sanity-check-the-installation) has been confirmed. 
+- You have followed the Ethernet setup as of the top of this page. 
+
+In this example it is assumed that you have followed the example to set your IP addresses.
+
+1. Connect your Flight controller via Ethernet
+2. Open **QGroundcontrol > Analyze Tools > MAVLink Console**
+3. Enter the command below to start micro_rtps client on your flight controller. Note that the remote IP here is your companion computer IP. This by default starts the micrortps_client connected to UDP ports 2019 and 2020. To make changes you can take a look at [RTPS guide](https://docs.px4.io/master/en/middleware/micrortps.html#client-px4-px4-autopilot)
+```
```

Comment:
If you want stuff to be under a list, it needs to be at the correct indentation level - 3 spaces.